src/server/search/search_family.cc

```diff
@@ -110,6 +117,47 @@ optional<SearchParams> ParseSearchParamsOrReply(CmdArgList args, ConnectionConte
       continue;
     }
 
+    // RETURN {num} [{ident} AS {name}...]
+    if (ArgS(args, i) == "RETURN") {
+      if (i + 1 >= args.size()) {
+        (*cntx)->SendError(kSyntaxErr);
+        return nullopt;
+      }
+
+      uint64_t num_fields = args.size();
+      if (!absl::SimpleAtoi(ArgS(args, i + 1), &num_fields)) {
+        (*cntx)->SendError(kSyntaxErr);
+        return nullopt;
+      }
+
+      i += 1;
+
+      alias_list = SearchParams::FieldAliasList{};
+      while (alias_list->size() < num_fields) {
+        if (i++ >= args.size()) {
+          (*cntx)->SendError(kSyntaxErr);
+          return nullopt;
+        }
+
+        string_view ident = ArgS(args, i);
+        string_view alias = ident;
+
+        if (i + 2 < args.size() && absl::EqualsIgnoreCase(ArgS(args, i + 1), "AS")) {
```

Comment:
We should also return a sytnax error if there exists an `i + 1 == AS` but there is no `I + 2`. So I think this should return an error:

`ft.search .... return 1 something as` I don't think it will now. 

Also consider adding a test case for that.

Reply:
Will be resolved with the parser, it handles it nicely